src/content/docs/en/basics/layouts.mdx

```diff
@@ -258,6 +258,52 @@ const { title } = Astro.props.frontmatter || Astro.props;
 </html>
 ```
 
+### Working with TypeScript
+
+Any astro layout can be modified to introduce typesafety & autocompletion by providing the types for your props:
+
+```astro ins={2-7} title="src/components/MyLayout.astro"
+---
+interface Props { 
+	title: string;
+  description: string;
+  publishDate: Date;
+  viewCount: number;
+}
+const { title, description, publishDate, viewCount } = Astro.props;
+---
+<html lang="en">
+  <head>
+    <meta charset="UTF-8">
+    <meta name="description" content={description}>
+    <title>{title}</title>
+  </head>
+  <body>
+    <header>
+      <p>Published on {publishDate}</p>
+      <p>Viewed by {viewCount} folks</p>
+    </header>
+    <main>
+      <slot />
+    </main>
+  </body>
+</html>
+```
+
+You can override Astro's default `Props` to handle situations where your layout recieves props of different types:
+
+```astro ": CustomProps" title="src/components/MyLayout.astro"
+---
+type CustomProps = { 
+	title: string;
+  description: string;
+  publishDate: Date;
+  viewCount: number;
+}
+const { title, description, publishDate, viewCount } : CustomProps = Astro.props.frontmatter || Astro.props;
+---
+```
+
```

Comment:
```suggestion
```
As I mentioned in the recent (second!) issue concerning this section, I think this section that was originally provided to serve one purpose for **v1** of Astro, long before content collections, has outlived its use. 

My suggestion here is to only add the new "Using TypeScript with layouts" section, and in fact remove lines 241-260 entirely! I don't think this section is helping people (vs causing problems) and I don't think with content collections now we really do recommend this anyway.